// @mysten/zksend@0.11.5/dist/cjs/wallet/channel/events.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var events_exports = {};
__export(events_exports, {
  StashedRequest: () => StashedRequest,
  StashedRequestData: () => StashedRequestData,
  StashedResponse: () => StashedResponse,
  StashedResponseData: () => StashedResponseData,
  StashedResponsePayload: () => StashedResponsePayload
});
module.exports = __toCommonJS(events_exports);
var import_valibot = require("valibot");
const StashedRequestData = (0, import_valibot.variant)("type", [
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("connect")
  }),
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("sign-transaction-block"),
    data: (0, import_valibot.string)("`data` is required"),
    address: (0, import_valibot.string)("`address` is required")
  }),
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("sign-personal-message"),
    bytes: (0, import_valibot.string)("`bytes` is required"),
    address: (0, import_valibot.string)("`address` is required")
  })
]);
const StashedRequest = (0, import_valibot.object)({
  id: (0, import_valibot.pipe)((0, import_valibot.string)("`id` is required"), (0, import_valibot.uuid)()),
  origin: (0, import_valibot.pipe)((0, import_valibot.string)(), (0, import_valibot.url)("`origin` must be a valid URL")),
  name: (0, import_valibot.optional)((0, import_valibot.string)()),
  payload: StashedRequestData
});
const StashedResponseData = (0, import_valibot.variant)("type", [
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("connect"),
    address: (0, import_valibot.string)()
  }),
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("sign-transaction-block"),
    bytes: (0, import_valibot.string)(),
    signature: (0, import_valibot.string)()
  }),
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("sign-personal-message"),
    bytes: (0, import_valibot.string)(),
    signature: (0, import_valibot.string)()
  })
]);
const StashedResponsePayload = (0, import_valibot.variant)("type", [
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("reject")
  }),
  (0, import_valibot.object)({
    type: (0, import_valibot.literal)("resolve"),
    data: StashedResponseData
  })
]);
const StashedResponse = (0, import_valibot.object)({
  id: (0, import_valibot.pipe)((0, import_valibot.string)(), (0, import_valibot.uuid)()),
  source: (0, import_valibot.literal)("zksend-channel"),
  payload: StashedResponsePayload
});
