// @mysten/zksend@0.11.5/dist/cjs/links/list-created-links.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var list_created_links_exports = {};
__export(list_created_links_exports, {
  listCreatedLinks: () => listCreatedLinks
});
module.exports = __toCommonJS(list_created_links_exports);
var import_bcs = require("@mysten/sui/bcs");
var import_graphql = require("@mysten/sui/graphql");
var import__ = require("@mysten/sui/graphql/schemas/2024.4");
var import_utils = require("@mysten/sui/utils");
var import_claim = require("./claim.js");
var import_zk_bag = require("./zk-bag.js");
const ListCreatedLinksQuery = (0, import__.graphql)(`
	query listCreatedLinks($address: SuiAddress!, $function: String!, $cursor: String) {
		transactionBlocks(
			last: 10
			before: $cursor
			filter: { signAddress: $address, function: $function, kind: PROGRAMMABLE_TX }
		) {
			pageInfo {
				startCursor
				hasPreviousPage
			}
			nodes {
				effects {
					timestamp
				}
				digest
				bcs
			}
		}
	}
`);
async function listCreatedLinks({
  address,
  cursor,
  network,
  contract = (0, import_zk_bag.getContractIds)(network),
  fetch: fetchFn,
  ...linkOptions
}) {
  const gqlClient = new import_graphql.SuiGraphQLClient({
    url: network === "testnet" ? "https://sui-testnet.mystenlabs.com/graphql" : "https://sui-mainnet.mystenlabs.com/graphql",
    fetch: fetchFn
  });
  const packageId = (0, import_utils.normalizeSuiAddress)(contract.packageId);
  const page = await gqlClient.query({
    query: ListCreatedLinksQuery,
    variables: {
      address,
      cursor,
      function: `${packageId}::zk_bag::new`
    }
  });
  const transactionBlocks = page.data?.transactionBlocks;
  if (!transactionBlocks || page.errors?.length) {
    throw new Error("Failed to load created links");
  }
  const links = (await Promise.all(
    transactionBlocks.nodes.map(async (node) => {
      if (!node.bcs) {
        return null;
      }
      const kind = import_bcs.bcs.SenderSignedData.parse((0, import_utils.fromBase64)(node.bcs))?.[0]?.intentMessage.value.V1.kind;
      if (!kind.ProgrammableTransaction) {
        return null;
      }
      const { inputs, commands } = kind.ProgrammableTransaction;
      const fn = commands.find(
        (command) => command.MoveCall?.package === packageId && command.MoveCall.module === "zk_bag" && command.MoveCall.function === "new"
      );
      if (!fn?.MoveCall) {
        return null;
      }
      const addressArg = fn.MoveCall.arguments[1];
      if (addressArg.$kind !== "Input") {
        throw new Error("Invalid address argument");
      }
      const input = inputs[addressArg.Input];
      if (!input.Pure) {
        throw new Error("Expected Address input to be a Pure value");
      }
      const address2 = import_bcs.bcs.Address.fromBase64(input.Pure.bytes);
      const link = new import_claim.ZkSendLink({
        network,
        address: address2,
        contract,
        isContractLink: true,
        ...linkOptions
      });
      await link.loadAssets();
      return {
        link,
        claimed: !!link.claimed,
        assets: link.assets,
        digest: node.digest,
        createdAt: node.effects?.timestamp
      };
    })
  )).reverse();
  return {
    cursor: transactionBlocks.pageInfo.startCursor,
    hasNextPage: transactionBlocks.pageInfo.hasPreviousPage,
    links: links.filter((link) => link !== null)
  };
}
