// @mysten/dapp-kit@0.14.24/dist/cjs/components/connect-modal/ConnectModal.css.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/connect-modal/ConnectModal.css.ts
var ConnectModal_css_exports = {};
__export(ConnectModal_css_exports, {
  backButtonContainer: () => backButtonContainer,
  closeButtonContainer: () => closeButtonContainer,
  content: () => content,
  overlay: () => overlay,
  selectedViewContainer: () => selectedViewContainer,
  title: () => title,
  viewContainer: () => viewContainer,
  walletListContainer: () => walletListContainer,
  walletListContainerWithViewSelected: () => walletListContainerWithViewSelected,
  walletListContent: () => walletListContent,
  whatIsAWalletButton: () => whatIsAWalletButton
});
module.exports = __toCommonJS(ConnectModal_css_exports);
var backButtonContainer = "ConnectModal_backButtonContainer__gz8z96";
var closeButtonContainer = "ConnectModal_closeButtonContainer__gz8z97";
var content = "ConnectModal_content__gz8z92";
var overlay = "ConnectModal_overlay__gz8z90";
var selectedViewContainer = "ConnectModal_selectedViewContainer__gz8z95";
var title = "ConnectModal_title__gz8z91";
var viewContainer = "ConnectModal_viewContainer__gz8z94";
var walletListContainer = "ConnectModal_walletListContainer__gz8z99";
var walletListContainerWithViewSelected = "ConnectModal_walletListContainerWithViewSelected__gz8z9a";
var walletListContent = "ConnectModal_walletListContent__gz8z98";
var whatIsAWalletButton = "ConnectModal_whatIsAWalletButton__gz8z93";
